### PR TITLE
fix(datetime): RTL will no longer infinitely scroll

### LINKED ISSUE #24472: bug: ion-datetime scrolls infinitely with dir=rtl
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

The calendar spins months backwards infinitely.
Peek 2021-12-22 19-18

### Expected Behavior

The calendar should stay still.

### Steps to Reproduce

Add `dir="rtl"` to `index.html`

### Code Reproduction URL

https://github.com/dima-gusyatiner/datetime-migration-samples/commit/99e44f3b507806445bc0174525f7322cbbeb76af

### Ionic Info

Ionic:
   Ionic CLI                     : 6.16.3 (/usr/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 6.0.0-rc.3
   @angular-devkit/build-angular : 12.1.4
   @angular-devkit/schematics    : 12.1.4
   @angular/cli                  : 12.1.4
   @ionic/angular-toolkit        : 4.0.0

System:
   NodeJS : v16.13.1 (/usr/bin/node)
   npm    : 8.3.0
   OS     : Linux 5.13

### Additional Information

Issue still persists on latest `angular` & `ionic` versions.

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

The date picker will infinitely scroll on initialization when using RTL. 

Issue Number: Resolves #24472

## What is the new behavior?

The calculations for scroll positioning will account for RTL. 

- Fixes the next/previous icons to flip in RTL

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

|Before|After|
|---|---|
|<video src="https://user-images.githubusercontent.com/13732623/147143117-574617b4-f66b-4a7e-bddb-e03b645b1d46.mp4"></video>| <video src="https://user-images.githubusercontent.com/13732623/147142970-d7cd570b-5080-4f6a-a02a-5400f8cc5c8f.mp4"></video> |

### CODE REVIEW

File: core/src/components/datetime/datetime.tsx [hunk 1]
```diff
@@ -995,9 +997,12 @@ export class Datetime implements ComponentInterface {
     const nextMonth = calendarBodyRef.querySelector('.calendar-month:last-of-type');
     if (!nextMonth) { return; }
 
+    const isRTL = document.dir === 'rtl';
```
Comment (liamdebeasi) [inline core/src/components/datetime/datetime.tsx:1000]: We need to check if `dir="rtl"` was set on the host element as well. Might be good to make this a utility that other components can use as well. Doing a quick search reveals that other components do not check the host when they probably should: https://github.com/ionic-team/ionic-framework/search?l=TSX&p=2&q=rtl

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

File: core/src/components/datetime/datetime.tsx (hunk 1 above)
Comment (sean-perkins) [reply core/src/components/datetime/datetime.tsx]: Added a new util for `isRTL`. 

How do you feel about splitting out a separate issue for each component that has rtl checks only for document? That way the change log will better reflect those changes to the util change and won't be tied to the datetime changes. If you feel good about that, I'm happy to create an issue for each.

Comment (liamdebeasi) [reply core/src/components/datetime/datetime.tsx]: Yep let's make some tech debt for that 👍

**Comment by amandaesmith3** [comment]
I'm still seeing the issue on Windows Chrome 🤔 Happens both when setting `dir="rtl"` on the `<html>`, or using `?rtl=true` in the URL. I added `console.log(isRTL(this.el));` at line 685 (right before setting `scrollLeft` and and `ioCallback`) and it's returning `false` for me.

**Comment by sean-perkins** [comment]
> I'm still seeing the issue on Windows Chrome 🤔 Happens both when setting `dir="rtl"` on the `<html>`, or using `?rtl=true` in the URL. I added `console.log(isRTL(this.el));` at line 685 (right before setting `scrollLeft` and and `ioCallback`) and it's returning `false` for me.

Ahh.. yeah I know the exact reason. The change to fix the lint error. One sec.

**Comment by sean-perkins** [comment]
Fixed that issue & added a test to cover that condition.

**Review by amandaesmith3 — APPROVED** [review]
There we go, looking great now 👍

File: core/src/utils/rtl/dir.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
+/**
+ * Returns `true` if the document or host element
+ * has a `dir` set to `rtl`. The host value will always
+ * take priority over the root document value.
+ */
+export const isRTL = (hostEl?: Pick<HTMLElement, 'dir'>) => {
+  if (hostEl) {
+    if (hostEl.dir !== '') {
+      return hostEl.dir.toLowerCase() === 'rtl';
```
Comment (liamdebeasi) [inline core/src/utils/rtl/dir.ts:10]: Should we be converting to lowercase? It looks `"Ltr"` should be treated as invalid looking at the accepted values: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir

Comment (sean-perkins) [reply core/src/utils/rtl/dir.ts:10]: I'm happy to swap that to explicit checks. Was siding on Dx, but if the browser will ignore any values that aren't `rtl | ltr | auto`, that isn't very helpful to be different.

Comment (liamdebeasi) [reply core/src/utils/rtl/dir.ts]: Oops I take it back. The browser `dir` is not case sensitive, so what you have currently aligns with how the browser behaves: https://codepen.io/liamdebeasi/pen/Jjrvgrp

Comment (sean-perkins) [reply core/src/utils/rtl/dir.ts:10]: Haha, that change is reverted 👍

**Review by liamdebeasi — APPROVED** [review]

**Comment by dviravr** [comment]
this isn't fix the problem for all cases.
if my app is in RTL mode, (on the 'ion-app') the infinity scroll is still appening. 
in addition, the arrows, are still upside down

**Comment by zhuhuanzi** [comment]
@sean-perkins This problem still exists in Android 9. Android 12 is ok.

**Comment by zhuhuanzi** [comment]
> @sean-珀金斯这个问题在Android 9中仍然存在。安卓12还可以。

 "@ionic/angular": "^6.0.12"

**Pull request merged**